dlc-test/src/test/scala/org/bitcoins/dlc/DLCClientTest.scala

```diff
@@ -208,11 +208,88 @@ class DLCClientTest extends BitcoinSAsyncTest {
       HashType.sigHashAll
     )
 
+    def validateFees(
+        fundingTx: Transaction,
+        closingTx: Transaction): Assertion = {
+      val builder = dlcOffer.dlcTxBuilder
+      val finalizer = builder.fundingTxBuilder.fundingTxFinalizer
+      val expectedFundingFee =
+        finalizer.offerFundingFee + finalizer.acceptFundingFee
+      val expectedClosingFee =
+        finalizer.offerFutureFee + finalizer.acceptFutureFee
+
+      val fundingInput =
+        (builder.offerFundingInputs ++ builder.acceptFundingInputs)
+          .map(_.output.value)
+          .sum
+      val fundingOutput = fundingTx.outputs.map(_.value).sum
+      val actualFundingFee = fundingInput - fundingOutput
+
+      val closingInput = fundingTx.outputs.head.value
+      val closingOutput = closingTx.outputs.map(_.value).sum
+      val actualClosingFee = closingInput - closingOutput
+
+      def feeRateBounds(
+          tx: Transaction,
+          actualFee: CurrencyUnit,
+          missingOutputBytes: Long = 0): (Double, Double) = {
+        val vbytesLower = Math.ceil(tx.weight / 4.0) + missingOutputBytes
+        val vbytesUpper =
+          vbytesLower + 2 // Have to add 2 to account for two parties rounding up
```

Comment:
Maybe in the spec we should say, only offerer or only acceptor rounds up

Reply:
eh
1) it is more complicated than just adding the vbyte, currently the spec has a nice almost simple computation and I don't want to add if/elses to it :P
2) this means one party pays some percentage of a vbyte more than the other on shared fields

Reply:
plus, re 2: both parties should be willing to go up so might as well